libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -606,17 +606,33 @@ bool IRGeneratorForStatements::visit(BinaryOperation const& _binOp)
 			solAssert(false, "Unknown comparison operator.");
 		define(_binOp) << expr << "\n";
 	}
-	else if (TokenTraits::isShiftOp(op) || op == Token::Exp)
+	else if (op == Token::Exp)
 	{
 		IRVariable left = convert(_binOp.leftExpression(), *commonType);
 		IRVariable right = convert(_binOp.rightExpression(), *type(_binOp.rightExpression()).mobileType());
-		if (op == Token::Exp)
+
+		if (auto literal = dynamic_cast<Literal const*>(&_binOp.leftExpression()))
+		{
+			RationalNumberType const* r = TypeProvider::rationalNumber(*literal);
+			solAssert(r, "Invalid literal as the base for exponentiation.");
+
```

Comment:
Please add an assertion that commonType is uint256 or int256 - and pass that type on to the helper function.

Reply:
We had a breaking change in 0.7.0 that the type of `literal ** exponent` is `uint256` or `int256` depending on the sign on `literal`.

Reply:
Ah ok